x/merkledb/db.go

```diff
@@ -1242,12 +1269,43 @@ func (db *merkleDB) getNode(key Key, hasValue bool) (*node, error) {
 	switch {
 	case db.closed:
 		return nil, database.ErrClosed
-	case key == Key{}:
-		return db.root, nil
 	case hasValue:
 		return db.valueNodeDB.Get(key)
+	default:
+		return db.intermediateNodeDB.Get(key)
+	}
+}
+
+func (db *merkleDB) getRoot() maybe.Maybe[*node] {
+	return db.root
+}
+
+func (db *merkleDB) Clear() error {
```

Comment:
Should we move the addition of `Clear` into a smaller PR? I don't think this really directly depends on the root change.

Reply:
https://github.com/ava-labs/avalanchego/pull/2277/files